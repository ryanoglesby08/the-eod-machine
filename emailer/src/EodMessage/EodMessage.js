import React from 'react'

import gql from 'graphql-tag'
import { Query } from 'react-apollo'

export const GET_EOD = gql`
  {
    eod {
      entries {
        category
        content
      }
    }
  }
`

const EodEmail = () => (
  <Query query={GET_EOD}>
    {({ data: { eod } }) => {
      const savedEntries = eod ? eod.entries : []

      return (
        <div>
          {savedEntries.map(({ category, content }) => (
            <div key={content}>
              {category}: {content}
            </div>
          ))}
        </div>
      )
    }}
  </Query>
)

export default EodEmail
