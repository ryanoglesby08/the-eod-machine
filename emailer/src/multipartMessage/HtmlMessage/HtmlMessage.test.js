import React from 'react'

import { MockedProvider } from 'react-apollo/lib/test-utils'
import { render, waitForElement } from 'react-testing-library'

import GET_EOD from '../getEodQuery'
import EodMessage from './HtmlMessage'

it('is an EOD message', async () => {
  const mocks = [
    {
      request: {
        query: GET_EOD,
      },
      result: {
        data: {
          eod: {
            entries: [
              { category: 'Category 1', content: 'some content' },
              { category: 'Category 1', content: 'more content' },
              { category: 'Category 2', content: 'even more content' },
            ],
          },
        },
      },
    },
  ]

  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <EodMessage />
    </MockedProvider>
  )

  await waitForElement()

  expect(container).toMatchSnapshot()
})
