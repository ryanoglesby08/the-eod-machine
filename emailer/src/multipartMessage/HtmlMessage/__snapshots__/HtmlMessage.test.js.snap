// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is an EOD message 1`] = `
<div>
  <h1>
    EOD Updates
  </h1>
  <div>
    <h2>
      Category 1
    </h2>
    <ul>
      <li>
        some content (Some author)
      </li>
      <li>
        more content (Another author)
      </li>
    </ul>
  </div>
  <div>
    <h2>
      Category 2
    </h2>
    <ul>
      <li>
        even more content (Some author)
      </li>
    </ul>
  </div>
  <div>
    Delivered by The EOD Machine
  </div>
</div>
`;

exports[`is an empty EOD message when there are no entries 1`] = `
<div>
  <h1>
    EOD Updates
  </h1>
  <div>
    No updates this time.
  </div>
  <div>
    Delivered by The EOD Machine
  </div>
</div>
`;
